### PR TITLE
Add interest page (v2)

### PR DESCRIPTION
**Pull request by bharathcs**

### CODE REVIEW

File: frontend/pages/interest.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        </Avatar>
+        <Stack spacing={1} marginBottom={2}>
+          <Typography component="h1" align="center" variant="h2">
+            Interested in starting a campaign?
+          </Typography>
+          <Typography variant="subtitle1" align="center">
+            Fill up the form below and we will get back to you!
+          </Typography>
+          <Typography variant="body2"></Typography>
+        </Stack>
```
Comment (zognin) [inline frontend/pages/interest.tsx:45]: This could be part of the form as it is the form title

File: frontend/pages/interest.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          <MailOutline />
+        </Avatar>
+        <Stack spacing={1} marginBottom={2}>
+          <Typography component="h1" align="center" variant="h2">
+            Interested in starting a campaign?
+          </Typography>
+          <Typography variant="subtitle1" align="center">
+            Fill up the form below and we will get back to you!
+          </Typography>
+          <Typography variant="body2"></Typography>
```
Comment (zognin) [inline frontend/pages/interest.tsx:44]: What does this do?

File: frontend/styles/interest/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  alignItems: 'center',
+};
+
+export const mailIconSx: SxProps = {
+  m: 1,
+  bgcolor: 'secondary.main',
+};
+
+export const amountButtonSx: SxProps = {
+  margin: 1,
```
Comment (zognin) [inline frontend/styles/interest/index.ts:15]: Maybe we can standardize to use `margin` or `m`. `m` is `margin` is it?

File: frontend/types/interest.ts [hunk 4]
```diff
@@ -13,8 +14,8 @@ export type Interest = {
   campaignName: string;
   campaignDescription: string;
   promisedAmount: number;
-  start: Date;
-  end: Date;
+  start: Moment | Date;
```
Comment (zognin) [inline frontend/types/interest.ts:17]: Do we use `Date` type anywhere?

File: frontend/utils/constants.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const DEFAULT_COUPON_DENOMINATION = 10;
```
Comment (zognin) [inline frontend/utils/constants.ts:1]: I think we decided to allow people to set this. But I'm okay with approving this first and just letting the admin set it when they create campaign.

**Review by zognin — COMMENTED** [review]
Frontend LGTM and reminds me to useField. There's this error from the backend even though I have a charity with id=1, it may be because the schema changed.
Screenshot 2022-10-12 at 6 08 24 PM

File: frontend/pages/interest.tsx (hunk 1 above)
Comment (zognin) [reply frontend/pages/interest.tsx:45]: Also nit but we can put a line break between same line tags for readability, but it's up to you

**Comment by zognin** [comment]
Why did yarn.lock change btw?

File: frontend/pages/interest.tsx (hunk 2 above)
Comment (bharathcs) [reply frontend/pages/interest.tsx:44]: thanks for the spot, removing.

File: frontend/types/interest.ts (hunk 4 above)
Comment (bharathcs) [reply frontend/types/interest.ts:17]: We do, with yup schema casting to Date at the end of validation. Our alternatives are to ignore type checking for start / end, use a moment yup schema (ugly, and requires more custom functions), or to transform the schema just before validation. 

For the last option, I prefer being explicit that the value stored can be either moment or date, so this is the soln I stuck with. Can take this offline to discuss more if you like.

**Review by zognin — APPROVED** [review]
LGTM thanks

**Pull request merged**